aptos-move/block-executor/src/captured_reads.rs

```diff
@@ -583,6 +625,25 @@ impl<T: Transaction> CapturedReads<T> {
         })
     }
 
+    pub(crate) fn validate_module_reads(
+        &self,
+        module_storage: &VersionedModuleStorage<T::Key>,
+        idx_to_validate: TxnIndex,
+    ) -> bool {
+        if self.speculative_failure {
+            return false;
+        }
+
+        // Only successful module storage reads are captured. In case there were no base value,
+        // and fetching it returned an error, the error is not recorded. Hence, it is safe here
+        // to simply treat such errors as a non-existent value.
+        self.module_storage_reads.iter().all(|(k, previous_read)| {
+            let result = module_storage.get(k, ShiftedTxnIndex::new(idx_to_validate));
```

Comment:
I think it's better to call with index and shift it inside implementation (also convention for existing interfaces afair)

Reply:
done